### PR TITLE
Feature/add emr pipeline status checker

### PR DESCRIPTION
**Pull request by ivche**
A util that implements : 
- fetch a cluster status
- fetch a pipeline status
- polls for a status every X seconds depending on the function passed to it

The intention is to import it and use it in the staging, sampling tools within different projects

### CODE REVIEW

File: aws_utils/emr/statuschecker.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class NoSuchActivityError(Exception):
+    """
+    A custom exception
+    """
```
Comment (lucasc896) [inline aws_utils/emr/statuschecker.py:13]: Add a 'pass' here

File: aws_utils/emr/statuschecker.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    # get the status of the pipeline
+    for pipeline in pipelines['pipelineIdList']:
+        if pipeline['name'] == pipeline_name:
+            pipeline_settings = boto3client_pipelines.describe_pipelines(pipelineIds=[pipeline['id']])
+            pipeline_fields = pipeline_settings['pipelineDescriptionList'][0]['fields']
+            pipeline_status = [item['stringValue'] for item in pipeline_fields if item['key'] == '@pipelineState']
+            if pipeline_status is not '':
+                return ''.join(pipeline_status)
+            else:
+                raise NoSuchActivityError('\tThe EMR status is empty')
```
Comment (joao-skimlinks) [inline aws_utils/emr/statuschecker.py]: Do you really need this initial tab in the error description?
Ideally if you need formatting, the code that displays it should do the formatting and not the error itself?

**Comment by robcowie** [comment]
Regarding the api, I'd prefer to see namespaces to clearly separate the 'get status' logic for EMR and AWS Pipeline. The `status checker` logic could still be shared. Something like this...

``` python
aws_utils.status.pipeline.get_status('pipeline_name')
aws_utils.status.pipeline.poll_for_status('pipeline_name', timeout=None)

aws_utils.status.emr.get_status('cluster_name')
aws_utils.status.emr.poll_for_status('cluster_name', timeout=None)
```

The two `poll_for_status` functions are simply conveniences that call `aws_utils.status.status_poller`, with the appropriate `get_status` function.

File: aws_utils/emr/statuschecker.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    pipelines = boto3client_pipelines.list_pipelines()
+
+    current_status_pipeline = ''
+    # get the status of the pipeline
+    for pipeline in pipelines['pipelineIdList']:
+        if pipeline['name'] == pipeline_name:
+            pipeline_settings = boto3client_pipelines.describe_pipelines(pipelineIds=[pipeline['id']])
+            pipeline_fields = pipeline_settings['pipelineDescriptionList'][0]['fields']
+            pipeline_status = [item['stringValue'] for item in pipeline_fields if item['key'] == '@pipelineState']
+            if pipeline_status is not '':
```
Comment (robcowie) [inline aws_utils/emr/statuschecker.py]: `pipeline_status` is a list. It can never be a zero-length string.

File: aws_utils/emr/statuschecker.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    Possible return values : ACTIVATING, CANCELED, CASCADE_FAILED, DEACTIVATING, FAILED, FINISHED, INACTIVE, PAUSED,
+                             PENDING, RUNNING, SHUTTING_DOWN, SKIPPED, TIMEDOUT, VALIDATING, WAITING_FOR_RUNNER, WAITING_ON_DEPENDENCIES
+    """
+    # create a client connection
+    boto3client_pipelines = boto3.client('datapipeline', region) #'us-east-1')
+    pipelines = boto3client_pipelines.list_pipelines()
+
+    current_status_pipeline = ''
+    # get the status of the pipeline
+    for pipeline in pipelines['pipelineIdList']:
```
Comment (robcowie) [inline aws_utils/emr/statuschecker.py:31]: It might be useful to check if the number of pipelines matched is 1, and raise exceptions if it is zero or >1.

**Comment by robcowie** [comment]
https://gist.github.com/robcowie/41e73cf69f5ed53b0dcf2798bb7ce142

Unfinished (i.e. there is some obvious code de-duplication to do). We'll talk about it tomorrow.

**Comment by ivche** [comment]
Moved to another branch, so I am closing the PR

**Pull request closed without merging**